plugins/modules/smart_proxy.py

```diff
@@ -127,7 +127,7 @@ def main():
         payload = {
             'id': smart_proxy['id'],
         }
-        entity = module.resource_action('capsule_content', 'lifecycle_environments', payload)
+        entity = module.resource_action('capsule_content', 'lifecycle_environments', payload, record_change=False)
```

Comment:
:+1:

Reply:
But still in "check_mode" when you just faked to create a new smart_proxy, but didn't, this call will fail.
And i just realize, our testing strategy does not cover this.

Reply:
What would need to be done to fix that? :-)

Reply:
You would hold the `fake_entity` from here:
https://github.com/theforeman/foreman-ansible-modules/blob/fe0f1d0d6f1ccd5cf2d00ff0794c7f5b940a6a32/plugins/module_utils/foreman_helper.py#L898

So you would not do that call but fake the result (as an empty list).

Reply:
I tried running the module with check_mode, it does what you describe:

```
--- before
+++ after
@@ -10,4 +10,4 @@
   - 3
   url: https://fm-proxy.dev.foreman.lab:9090
 smart_proxy_content/lifecycle_environment_ids:
-- []
+- - 4
```

Reply:
Hmm, beginning to think, that our tests actually capture this. ;)
Let's see whether the check_mode test will pass.